integration/internal/process.go

```diff
@@ -16,7 +17,13 @@ import (
 
 type ProcessState struct {
 	DefaultedProcessInput
-	Session      *gexec.Session
+	Session *gexec.Session
+	// Healthcheck Endpoint. If we get http.StatusOK from this endpoint, we
+	// assume the process is ready to operate. E.g. "/healthz". If this is set,
+	// we ignore StartMessage.
+	HealthCheckEndpoint string
+	// Message to wait for on stderr. If we recieve this message, we assume the
```

Comment:
Why is it desirable to still support waiting for a message on stderr?  All kube api servers and controllers are likely to have healthcheck endpoints given that is how pod health is generally determined.

Reply:
- We didn't port all the things, only the APIServer, to use that new method
- We are not sure if `etcd` has an health check API (it probably has)
- We know that the virtual kubelet with the mock provider does not expose its health via API (see #41)
- I think it does not hurt or adds too much more complexity
- If we find that all the things we want to support have an API and all the processes are ported to use that new mechanism, we are happy to phase out the detection via waiting for a string on stderr.

Reply:
Consider updating the comment for ``StartMessage`` discouraging its use in favor of a health check.  It's not the complexity I'm worried about, rather that anyone naively relying on a startup message appearing in stderr is likely to run into the same type of problem that prompted you to create this PR.

Reply:
Mark as deprecated? (https://blog.golang.org/godoc-documenting-go-code)